### PR TITLE
Remove support of 'years' and 'months' timeunits from CRD

### PR DESCRIPTION
**Pull request by btaani**
### Description
This PR removes the support of `y` and `M` time units that are specified in Elasticsearch CRD (`yMwdhHms`).

/cc @sasagarw 
/assign @periklis 

### Links
- JIRA: [LOG-2418](https://issues.redhat.com/browse/LOG-2418)

### CODE REVIEW

**Comment by periklis** [comment]
This change looks not backward compatible

**Comment by btaani** [comment]
Yes we never supported months or years, it was just there in the CRD but whenever, for example `maxAge: 1M` is used, EO pod displays an error message that this time unit is not supported.

Also, should I remove this from the test file?
https://github.com/openshift/elasticsearch-operator/blob/d20f9cbc4995761090daf1b994642ca7088188f2/internal/indexmanagement/converters_test.go#L82

File: bundle/manifests/elasticsearch-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -457,7 +457,9 @@ spec:
           - watch
         serviceAccountName: elasticsearch-operator
       deployments:
-      - name: elasticsearch-operator
+      - label:
+          name: elasticsearch-operator
```
Comment (Red-GV) [inline bundle/manifests/elasticsearch-operator.clusterserviceversion.yaml:461]: This is unrelated to the bug from operator-sdk 18.1 right?

**Review by Red-GV — COMMENTED** [review]
One quick question. Otherwise, looks good. I think we can keep the test case.

File: bundle/manifests/elasticsearch-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (btaani) [reply bundle/manifests/elasticsearch-operator.clusterserviceversion.yaml:461]: So I had to update operator-sdk to 1.19.1 in order to remake the manifests, and this was generated after `make bundle` and `make generate`
this is probably in the wrong PR

Comment (Red-GV) [reply bundle/manifests/elasticsearch-operator.clusterserviceversion.yaml:461]: Hmm, okay. Maybe let's remove this part for now and then we might be all good?

**Comment by periklis** [comment]
> Yes we never supported months or years, it was just there in the CRD but whenever, for example `maxAge: 1M` is used, EO pod displays an error message that this time unit is not supported.
> 
> 
> 
> Also, should I remove this from the test file?
> 
> https://github.com/openshift/elasticsearch-operator/blob/d20f9cbc4995761090daf1b994642ca7088188f2/internal/indexmanagement/converters_test.go#L82
> 
> 
> 
> 
> 
> 

Regardless if the two units ever reconciled any jobs properly, the compatibility issue here affects the etcd store. If a user has stored an instance of the Elasticsearch CRD with these values and we remove them, we won't be able to read it back. Removing values here is like removing fields. @xperimental WDYT?

**Comment by xperimental** [comment]
> If a user has stored an instance of the Elasticsearch CRD with these values and we remove them, we won't be able to read it back. Removing values here is like removing fields.

As far as I remember, the validation only happens on the write path, so the CRD should still be readable, but it should no longer be possible to write new resources or update existing ones which have values that don't match the new pattern. While it is true, that removing valid values is a breaking change and so would need a versioned changed, I feel like this does not apply to values that were never usable / resulted in an error.
